clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -196,6 +197,44 @@ bool Sema::isKnownGoodSYCLDecl(const Decl *D) {
   return false;
 }
 
+bool isArraySizedZero(QualType Ty) {
+  if (const auto *CATy = dyn_cast<ConstantArrayType>(Ty)) {
+    const llvm::APInt size = CATy->getSize();
+    return size == 0;
+  }
+  return false;
+}
+
+void Sema::CheckVarDeclOKIfInKernel(VarDecl *var) {
+  // not all variable types supported in kernel contexts
+  // if not we record a deferred diagnostic.
+  if (getLangOpts().SYCLIsDevice) {
```

Comment:
Since this is only being checked in SYCL mode, just do an assert on this instead.

Reply:
done.
Q: to we write test cases for the assertions?